addons/hr_expense/models/hr_expense.py

```diff
@@ -181,7 +181,7 @@ def create_expense_from_attachments(self, attachment_ids=None, view_type='tree')
         if any(attachment.res_id or attachment.res_model != 'hr.expense' for attachment in attachments):
             raise UserError(_("Invalid attachments!"))
 
-        product = self.env['product.product'].search([('default_code', '=', 'EXP_GEN')])
+        product = self.env['product.product'].search([('default_code', '=', 'EXP_GEN')]) or self.env['product.product'].search([('can_be_expensed', '=', True)], limit = 1)
```

Comment:
Hi ! Thanks for the feedback. As a call to search on the product.product is quite heavy (See: https://github.com/odoo/odoo/blame/master/addons/product/models/product.py#L498), you should mix them together. 

Have a nice day.

Reply:
Hi! no problem.
I just do one search now and then filter the results. Is it better like that ? 

nice day too.

Reply:
Sounds good. Thanks.